examples/api/lib/widgets/interactive_viewer/interactive_viewer.builder.0.dart

```diff
@@ -150,38 +104,41 @@ typedef _CellBuilder = Widget Function(
 
 class _TableBuilder extends StatelessWidget {
   const _TableBuilder({
-    required this.rowCount,
-    required this.columnCount,
     required this.cellWidth,
+    required this.cellHeight,
+    required this.viewport,
     required this.builder,
-  })  : assert(rowCount > 0),
-        assert(columnCount > 0);
+  });
 
-  final int rowCount;
-  final int columnCount;
   final double cellWidth;
+  final double cellHeight;
+  final Rect viewport;
   final _CellBuilder builder;
 
   @override
   Widget build(BuildContext context) {
-    return Table(
-      // ignore: prefer_const_literals_to_create_immutables
-      columnWidths: <int, TableColumnWidth>{
-        for (int column = 0; column < columnCount; column++)
-          column: FixedColumnWidth(cellWidth),
-      },
-      // ignore: prefer_const_literals_to_create_immutables
-      children: <TableRow>[
-        for (int row = 0; row < rowCount; row++)
-          // ignore: prefer_const_constructors
-          TableRow(
-            // ignore: prefer_const_literals_to_create_immutables
-            children: <Widget>[
-              for (int column = 0; column < columnCount; column++)
-                builder(context, row, column),
-            ],
-          ),
-      ],
-    );
+    int firstRow = (viewport.top / cellHeight).floor();
+    int lastRow = (viewport.bottom / cellHeight).ceil();
+    int firstCol = (viewport.left / cellWidth).floor();
+    int lastCol = (viewport.right / cellWidth).ceil();
+
+    int totalCells = (lastRow - firstRow) * (lastCol - firstCol);
+    debugPrint("Total cells: $totalCells");
+
+    return SizedBox(
+        // Stack needs constraints, even though we then Clip.none outside of them.
```

Comment:
This interaction is also kind of weird to deal with.

Reply:
InteractiveViewer.builder always sets [constrained](https://master-api.flutter.dev/flutter/widgets/InteractiveViewer/constrained.html) to false, giving infinite constraints to the child.